### PR TITLE
Bump Lighthouse, drop the CoreDNS dependencies

### PR DESCRIPTION
**Pull request by skitt**
This drops the dependency on CoreDNS and allows us to remove the
pinning on most Kubernetes dependencies. We still need to pin
client-go, because prometheus-operator pulls in v12.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

**Comment by tpantelis** [comment]
We need to downgrade `github.com/go-logr/logr` to 0.4.0.

File: go.mod [hunk 1]
```diff
@@ -62,11 +65,6 @@ require (
 // Pinned to kubernetes-1.21.11
 replace (
 	github.com/go-logr/logr => github.com/go-logr/logr v0.4.0
```
Comment (tpantelis) [inline go.mod]: I think we can specify this directly on L13 instead of replace?

File: go.mod [hunk 2]
```diff
@@ -62,11 +65,6 @@ require (
...
-	k8s.io/api => k8s.io/api v0.21.11
-	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.21.11
-	k8s.io/apimachinery => k8s.io/apimachinery v0.21.11
 	k8s.io/client-go => k8s.io/client-go v0.21.11
-	k8s.io/cloud-provider => k8s.io/cloud-provider v0.21.11
 	k8s.io/klog/v2 => k8s.io/klog/v2 v2.9.0
```
Comment (tpantelis) [inline go.mod:68]: Do we still need this?

Comment (skitt) [reply go.mod:68]: I think so, but let’s try without...

Comment (skitt) [reply go.mod:68]: Yup, we do still need it.

**Review by dfarrell07 — APPROVED** [review]

**Review by tpantelis — APPROVED** [review]

**Pull request merged**